File: JAVA/Program/Pembeli.java
import java.util.ArrayList;

public class Pembeli extends User {
    private String idPembeli, alamat;
    private ArrayList<Produk> keranjang = new ArrayList<>();

    public Pembeli(String idUser, String nama, String email, String password, String noHp,
                   String idPembeli, String alamat) {
        super(idUser, nama, email, password, noHp);
        this.idPembeli = idPembeli;
        this.alamat = alamat;
    }

    public void tambahKeKeranjang(Produk p) {
        keranjang.add(p);
    }

    @Override
    public void tampilkanInfo() {
        super.tampilkanInfo();
        System.out.println("ID Pembeli : " + idPembeli);
        System.out.println("Alamat     : " + alamat);
        System.out.println("=== Keranjang ===");
        for (Produk p : keranjang) {
            p.tampilkanInfo();
            System.out.println("-----------------");
        }
    }
}

File: PYTHON/Program/Pembeli.py
from User import User

class Pembeli(User):
    def __init__(self, id_user, nama, email, password, no_hp,
                 id_pembeli, alamat):
        super().__init__(id_user, nama, email, password, no_hp)
        self.id_pembeli = id_pembeli
        self.alamat = alamat
        self.keranjang = []

    def tambah_ke_keranjang(self, produk):
        self.keranjang.append(produk)

    def tampilkan_info(self):
        super().tampilkan_info()
        print(f"ID Pembeli : {self.id_pembeli}")
        print(f"Alamat     : {self.alamat}")
        print("=== Keranjang ===")
        for p in self.keranjang:
            p.tampilkan_info()
            print("-----------------")
